src/Integration/NewConnectedMode/ConfigurationProvider.cs

```diff
@@ -65,19 +68,18 @@ public bool WriteConfiguration(BindingConfiguration configuration)
                 throw new ArgumentNullException(nameof(configuration));
             }
 
-            // For legacy mode, we need to write the configuration to
-            // disk, as well as storing it in the in-memory provider
-            if (configuration.Mode == SonarLintMode.LegacyConnected)
+            switch (configuration.Mode)
             {
-                bool success = legacySerializer.WriteSolutionBinding(configuration.Project) != null;
-                if (success)
-                {
-                    wrappedProvider.WriteConfiguration(configuration);
-                }
-                return success;
+                case SonarLintMode.LegacyConnected:
+                    return legacySerializer.WriteSolutionBinding(configuration.Project) != null;
+                case SonarLintMode.Connected:
+                    return newConnectedModeSerializer.WriteSolutionBinding(configuration.Project) != null;
+                case SonarLintMode.Standalone:
+                    throw new InvalidOperationException(Strings.Bind_CannotSaveStandaloneConfiguration);
+                default:
+                    Debug.Fail("Unrecognised write mode");
+                    return false;
```

Comment:
Unfortunately, the removal of the _InMemoryConfigurationProvider_ means that every time a component asks for the current configuration, we'll attempt to read from disk (and in the case of standalone mode, we'll tried to read from two different locations, but find nothing).

I'll fix this if I have time, probably by making this component store the config in-memory, and adding a "refresh"  option that checks for the files on disk. The "refresh" would be called whenever a solution is opened or closed. That should be enough to keep the in-memory state in sync.